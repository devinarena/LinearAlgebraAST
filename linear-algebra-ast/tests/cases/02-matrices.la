let A = [1];
let another = [1 2|3 4];
let B = [1 2 3|4 5 6|7 8 9];

let C = [3];
let D = [1 0 0|0 1 0|0 0 1];

let vector = [2|1];

print A;
newline;
print another;
newline;
print B;
newline;

print A * C;
newline;
print B * D;
newline;
print another * vector;
newline;

print D^99;
newline;
print another^2;
newline;
print -B;