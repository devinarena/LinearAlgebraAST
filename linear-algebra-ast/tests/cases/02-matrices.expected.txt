TOKEN_LET - let - 1
TOKEN_IDENTIFIER - A - 1
TOKEN_EQUAL - = - 1
TOKEN_LEFT_BRACKET - [ - 1
TOKEN_NUMBER - 1 - 1
TOKEN_RIGHT_BRACKET - ] - 1
TOKEN_SEMICOLON - ; - 1
TOKEN_LET - let - 2
TOKEN_IDENTIFIER - another - 2
TOKEN_EQUAL - = - 2
TOKEN_LEFT_BRACKET - [ - 2
TOKEN_NUMBER - 1 - 2
TOKEN_NUMBER - 2 - 2
TOKEN_PIPE - | - 2
TOKEN_NUMBER - 3 - 2
TOKEN_NUMBER - 4 - 2
TOKEN_RIGHT_BRACKET - ] - 2
TOKEN_SEMICOLON - ; - 2
TOKEN_LET - let - 3
TOKEN_IDENTIFIER - B - 3
TOKEN_EQUAL - = - 3
TOKEN_LEFT_BRACKET - [ - 3
TOKEN_NUMBER - 1 - 3
TOKEN_NUMBER - 2 - 3
TOKEN_NUMBER - 3 - 3
TOKEN_PIPE - | - 3
TOKEN_NUMBER - 4 - 3
TOKEN_NUMBER - 5 - 3
TOKEN_NUMBER - 6 - 3
TOKEN_PIPE - | - 3
TOKEN_NUMBER - 7 - 3
TOKEN_NUMBER - 8 - 3
TOKEN_NUMBER - 9 - 3
TOKEN_RIGHT_BRACKET - ] - 3
TOKEN_SEMICOLON - ; - 3
TOKEN_LET - let - 5
TOKEN_IDENTIFIER - C - 5
TOKEN_EQUAL - = - 5
TOKEN_LEFT_BRACKET - [ - 5
TOKEN_NUMBER - 3 - 5
TOKEN_RIGHT_BRACKET - ] - 5
TOKEN_SEMICOLON - ; - 5
TOKEN_LET - let - 6
TOKEN_IDENTIFIER - D - 6
TOKEN_EQUAL - = - 6
TOKEN_LEFT_BRACKET - [ - 6
TOKEN_NUMBER - 1 - 6
TOKEN_NUMBER - 0 - 6
TOKEN_NUMBER - 0 - 6
TOKEN_PIPE - | - 6
TOKEN_NUMBER - 0 - 6
TOKEN_NUMBER - 1 - 6
TOKEN_NUMBER - 0 - 6
TOKEN_PIPE - | - 6
TOKEN_NUMBER - 0 - 6
TOKEN_NUMBER - 0 - 6
TOKEN_NUMBER - 1 - 6
TOKEN_RIGHT_BRACKET - ] - 6
TOKEN_SEMICOLON - ; - 6
TOKEN_LET - let - 8
TOKEN_IDENTIFIER - vector - 8
TOKEN_EQUAL - = - 8
TOKEN_LEFT_BRACKET - [ - 8
TOKEN_NUMBER - 2 - 8
TOKEN_PIPE - | - 8
TOKEN_NUMBER - 1 - 8
TOKEN_RIGHT_BRACKET - ] - 8
TOKEN_SEMICOLON - ; - 8
TOKEN_PRINT - print - 10
TOKEN_IDENTIFIER - A - 10
TOKEN_SEMICOLON - ; - 10
TOKEN_NEWLINE - newline - 11
TOKEN_SEMICOLON - ; - 11
TOKEN_PRINT - print - 12
TOKEN_IDENTIFIER - another - 12
TOKEN_SEMICOLON - ; - 12
TOKEN_NEWLINE - newline - 13
TOKEN_SEMICOLON - ; - 13
TOKEN_PRINT - print - 14
TOKEN_IDENTIFIER - B - 14
TOKEN_SEMICOLON - ; - 14
TOKEN_NEWLINE - newline - 15
TOKEN_SEMICOLON - ; - 15
TOKEN_PRINT - print - 17
TOKEN_IDENTIFIER - A - 17
TOKEN_STAR - * - 17
TOKEN_IDENTIFIER - C - 17
TOKEN_SEMICOLON - ; - 17
TOKEN_NEWLINE - newline - 18
TOKEN_SEMICOLON - ; - 18
TOKEN_PRINT - print - 19
TOKEN_IDENTIFIER - B - 19
TOKEN_STAR - * - 19
TOKEN_IDENTIFIER - D - 19
TOKEN_SEMICOLON - ; - 19
TOKEN_NEWLINE - newline - 20
TOKEN_SEMICOLON - ; - 20
TOKEN_PRINT - print - 21
TOKEN_IDENTIFIER - another - 21
TOKEN_STAR - * - 21
TOKEN_IDENTIFIER - vector - 21
TOKEN_SEMICOLON - ; - 21
TOKEN_NEWLINE - newline - 22
TOKEN_SEMICOLON - ; - 22
TOKEN_PRINT - print - 24
TOKEN_IDENTIFIER - D - 24
TOKEN_CARET - ^ - 24
TOKEN_NUMBER - 99 - 24
TOKEN_SEMICOLON - ; - 24
TOKEN_NEWLINE - newline - 25
TOKEN_SEMICOLON - ; - 25
TOKEN_PRINT - print - 26
TOKEN_IDENTIFIER - another - 26
TOKEN_CARET - ^ - 26
TOKEN_NUMBER - 2 - 26
TOKEN_SEMICOLON - ; - 26
TOKEN_NEWLINE - newline - 27
TOKEN_SEMICOLON - ; - 27
TOKEN_PRINT - print - 28
TOKEN_MINUS - - - 28
TOKEN_IDENTIFIER - B - 28
TOKEN_SEMICOLON - ; - 28
TOKEN_EOF -  - 28
| 1 |

| 1 2 |
| 3 4 |

| 1 2 3 |
| 4 5 6 |
| 7 8 9 |

| 3 |

| 1 2 3 |
| 4 5 6 |
| 7 8 9 |

| 4 |
| 10 |

| 1 0 0 |
| 0 1 0 |
| 0 0 1 |

| 7 10 |
| 15 22 |

| -1 -2 -3 |
| -4 -5 -6 |
| -7 -8 -9 |
